Original patch from: ../4.3.2/230-pr34571.patch

-= BEGIN original header =-
Original patch from gentoo: gentoo/src/patchsets/gcc/4.3.1/gentoo/64_all_gcc43-pr34571.patch
http://gcc.gnu.org/PR34571

2007-12-26  Rask Ingemann Lambertsen  <rask@sygehus.dk>

	PR target/34571
	* config/alpha/alpha.c (alpha_cannot_force_const_mem): Use
	symbolic_operand.
	* varasm.c (output_constant_pool_1): Fix typo.

-= END original header =-

diff -durN gcc-4.3.3.orig/gcc/config/alpha/alpha.c gcc-4.3.3/gcc/config/alpha/alpha.c
--- gcc-4.3.3.orig/gcc/config/alpha/alpha.c	2008-09-09 01:16:55.000000000 +0200
+++ gcc-4.3.3/gcc/config/alpha/alpha.c	2009-01-27 22:25:29.000000000 +0100
@@ -1113,8 +1113,7 @@
 static bool
 alpha_cannot_force_const_mem (rtx x)
 {
-  enum rtx_code code = GET_CODE (x);
-  return code == SYMBOL_REF || code == LABEL_REF || code == CONST;
+  return symbolic_operand (x, GET_MODE (x));
 }
 
 /* We do not allow indirect calls to be optimized into sibling calls, nor
